includes/csv_arrow_ingest.hpp

```diff
@@ -0,0 +1,9 @@
+#include <arrow/csv/api.h>
+
+std::shared_ptr<arrow::Table>
+ReadEncryptedCsv(const std::string &filename, const std::string *decryption_key,
```

Comment:
Because references can't be null. 
`get_arrow_convert_options` checks whether `utf8_columns` is null because in 2 out of 3 file types, default CSV reading behavior is needed, but in the remaining case (update files), all columns have to be read in as text.
Similar thing with `decryption_key` -- `process_file()` checks if it's null to decide whether a file is plain-text or encrypted/compressed.

Reply:
We need to make a distinction between empty container (string/vector) and null?

Reply:
I guess empty should be enough of a signal.